### ISSUE TITLE
Filter tags when using the `logbooks` param in `GET /entries?logbooks={logbookId}`

### ISSUE THREAD

**Issue by ocboogie** [eed-web-application/elog-plus#248]
When requesting `GET /entries?logbooks=6536f0b5fbf92e44a767571c` where `6536f0b5fbf92e44a767571c` is the ID of PEP, the backend includes tags from other logbooks (see JSON dump below). This design could be okay, however the field `logbooks` is filtered such that only `PEP` is included. I'd like consistency here where either both the `logbooks` and `tags` fields are filtered based on the `logbooks` param, or neither `logbooks` or `tags` are filtered. I think it's a better a REST API design to have neither be filtered, since query params shouldn't modify the resources (in this case, entries) returned, but rather just filter them out (see [this](https://stackoverflow.com/questions/4024271/rest-api-best-practices-where-to-put-parameters)). However, it doesn't make a difference for me on the frontend, since when using the `logbooks` param I don't use any information from the other logbooks. So, for the sake of simplicity and reducing bandwidth over the network, I think filtering out tags that aren't included in the `logbooks` param is best. 

TLDR: filter out tags where `tag.logbook.id` are not in the `logbooks` query param.
```
{
  "errorCode": 0,
  "payload": [
    {
      "id": "6538479f5ec1e72404bf3116",
      "logbooks": [
        {
          "id": "6536f0b5fbf92e44a767571c",
          "name": "pep"
        }
      ],
      "title": "Sectors 21 and 22 klystron filament heaters are OFF",
      "loggedBy": "Paul Stiles",
      "tags": [
        {
          "id": "076dfad9-0fc6-4839-8c42-d16e1e9ab051",
          "name": "facet",
          "logbook": {
            "id": "6536f0b5fbf92e44a767572d",
            "name": "amrf"
          }
        },
        {
          "id": "dce26dca-142d-40eb-9d1d-d480d8dc57f9",
          "name": "lcls",
          "logbook": {
            "id": "6536f0b5fbf92e44a767572d",
            "name": "amrf"
          }
        },
        {
          "id": "cda25a68-8dca-48c9-9d35-34e0e9f454d7",
          "name": "facet",
          "logbook": {
            "id": "6536f0b5fbf92e44a7675736",
            "name": "facilities"
          }
        },
        {
          "id": "4906a3c2-2cd1-426e-b4eb-eeb23c6cb814",
          "name": "lcls",
          "logbook": {
            "id": "6536f0b5fbf92e44a7675736",
            "name": "facilities"
          }
        },
        {
          "id": "a322161d-e930-4ad4-b481-7aa10be41996",
          "name": "facet",
          "logbook": {
            "id": "6536f0b5fbf92e44a7675730",
            "name": "pem"
          }
        },
        {
          "id": "fe1151d3-12a3-4b07-b9ac-d9e4bfcd1561",
          "name": "lcls",
          "logbook": {
            "id": "6536f0b5fbf92e44a7675730",
            "name": "pem"
          }
        },
        {
          "id": "13d9250a-7788-4c5e-a839-b36e1ccc2c86",
          "name": "facet",
          "logbook": {
            "id": "6536f0b5fbf92e44a767571c",
            "name": "pep"
          }
        },
        {
          "id": "e9982daa-cc8f-4946-84d3-6c45c79fecca",
          "name": "lcls",
          "logbook": {
            "id": "6536f0b5fbf92e44a767571c",
            "name": "pep"
          }
        }
      ],
      "attachments": [],
      "shifts": [],
      "references": [],
      "referencedBy": [],
      "loggedAt": "2022-01-03T18:02:52",
      "eventAt": "2022-01-03T17:26:05"
    }
  ]
}
```

**Comment by bisegni** [comment]
it's bug, the tag should be filtered too, thank for reporting
